com.unity.render-pipelines.universal/Runtime/2D/Passes/Utility/RendererLighting.cs

```diff
@@ -199,14 +215,19 @@ private static void RenderLightSet(IRenderPass2D pass, RenderingData renderingDa
                 var batchedLights = 0;
 
                 // Add lights to our batch until the number of shadow textures reach the maxShadowTextureCount
-                var shadowLightCount = 0;
+                int shadowLightCount = 0;
                 while (batchedLights < remainingLights && shadowLightCount < maxShadowLightCount)
                 {
-                    var light = lights[lightIndex + batchedLights];
-                    if (light.shadowsEnabled && light.shadowIntensity > 0 && light.IsLitLayer(layerToRender))
+                    int curLightIndex = lightIndex + batchedLights;
+                    var light = lights[curLightIndex];
+                    if (CanCastShadows(light, layerToRender))
                     {
-                        ShadowRendering.PrerenderShadows(pass, renderingData, cmd, layerToRender, light, shadowLightCount, light.shadowIntensity);
-                        shadowLightCount++;
+                        doesLightAtIndexHaveShadows[curLightIndex] = false;
```

Comment:
The default values of the array should already be `false`. Do you still need this line?